src/dhnx/gistools/geometry_operations.py

```diff
@@ -364,6 +364,9 @@ def weld_segments(
         gdf_line_net_new = _weld_segments(
             gdf_line_net_new, gdf_line_gen, gdf_line_houses, debug_plotting
         )
+        if len(gdf_line_net_new) == 0:
+            gdf_line_net_new = gdf_line_net_last
+            break
```

Comment:
Can you please add a comment when this might happen?

Reply:
If I remember correctly, I had a case where ``weld_segments`` would just delete all lines in a network. Adding that change did not solve my actual problem (because it only returns the previous state of an almost deleted network) but it simplified searching for the root of the problem by not breaking the following code (which would throw other errors with an empty network).
I must admit that I did not store the problematic input network and cannot remember the details. As I said, it was a rare edge case.